x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicy.java

```diff
@@ -79,6 +98,11 @@ public LifecyclePolicy(StreamInput in) throws IOException {
         type = in.readNamedWriteable(LifecycleType.class);
         name = in.readString();
         phases = Collections.unmodifiableMap(in.readMap(StreamInput::readString, Phase::new));
+        if (in.readBoolean()) {
+            this.metadata = in.readMap();
+        } else {
+            this.metadata = null;
+        }
```

Comment:
This needs some version control logic, something like:

```java
if (in.getVersion().onOrAfter(Version.V8_0_0)) {
    this.metadata = in.readMap();
} else {
    this.metadata = null;
}
```

(`readMap()` and `writeMap(...)` can return null and handle null values)